tests/test_opentelemetry.py

```diff
@@ -0,0 +1,143 @@
+from typing import AsyncGenerator
+
+import pytest
+from opentelemetry.sdk.trace.export.in_memory_span_exporter import InMemorySpanExporter
+from opentelemetry.trace import StatusCode
+
+from replit_river.client import Client
+from replit_river.error_schema import RiverError
+from tests.conftest import deserialize_error, deserialize_response, serialize_request
+from tests.river_fixtures.logging import NoErrors
+
+
+@pytest.mark.asyncio
+async def test_rpc_method_span(
+    client: Client, span_exporter: InMemorySpanExporter
+) -> None:
+    response = await client.send_rpc(
+        "test_service",
+        "rpc_method",
+        "Alice",
+        serialize_request,
+        deserialize_response,
+        deserialize_error,
+    )
+    assert response == "Hello, Alice!"
+    spans = span_exporter.get_finished_spans()
+    assert len(spans) == 1
+    assert spans[0].name == "river.client.rpc.test_service.rpc_method"
+
+
+@pytest.mark.asyncio
+async def test_upload_method_span(
+    client: Client, span_exporter: InMemorySpanExporter
+) -> None:
+    async def upload_data() -> AsyncGenerator[str, None]:
+        yield "Data 1"
+        yield "Data 2"
+        yield "Data 3"
+
+    response = await client.send_upload(
+        "test_service",
+        "upload_method",
+        "Initial Data",
+        upload_data(),
+        serialize_request,
+        serialize_request,
+        deserialize_response,
+        deserialize_error,
+    )
+    assert response == "Uploaded: Initial Data, Data 1, Data 2, Data 3"
+    spans = span_exporter.get_finished_spans()
+    assert len(spans) == 1
+    assert spans[0].name == "river.client.upload.test_service.upload_method"
+
+
+@pytest.mark.asyncio
+async def test_subscription_method_span(
+    client: Client, span_exporter: InMemorySpanExporter
+) -> None:
+    async for response in client.send_subscription(
+        "test_service",
+        "subscription_method",
+        "Bob",
+        serialize_request,
+        deserialize_response,
+        deserialize_error,
+    ):
+        assert isinstance(response, str)
+        assert "Subscription message" in response
```

Comment:
Far from me to ask for _more_ convoluted tests, but am I correct in presuming that it's not possible to test exception propagation within the context of OTEL here?

Reply:
Say more? The last test in this file does test for errors coming from a stream rpc, I can add erorr tests for the other rpcs though!

Reply:
I completely missed that that test was testing what I was looking for. I expected it to be of the form

```
    async def stream_data() -> AsyncGenerator[str, None]:
        raise RiverError(...)
        yield "nope"
```

The circuitous path back towards a raised error threw me off. If that's what's required, I'm happy to adjust my expectations and bikeshed locally one of these weekends 😉

Reply:
Ah thats because `stream_data` is the client->server stream, there's also the server->client stream which is the one that has the error and the path we want to test.

Reply:
Makes sense. I am mollified.